core/src/main/java/org/apache/iceberg/variants/PrimitiveWrapper.java

```diff
@@ -47,17 +48,23 @@ class PrimitiveWrapper<T> implements VariantPrimitive<T> {
   private static final byte BINARY_HEADER = VariantUtil.primitiveHeader(Primitives.TYPE_BINARY);
   private static final byte STRING_HEADER = VariantUtil.primitiveHeader(Primitives.TYPE_STRING);
 
-  private final Variants.PhysicalType type;
+  private final PhysicalType type;
   private final T value;
   private ByteBuffer buffer = null;
 
-  PrimitiveWrapper(Variants.PhysicalType type, T value) {
-    this.type = type;
+  PrimitiveWrapper(PhysicalType type, T value) {
+    if (value instanceof Boolean
```

Comment:
nit: seems I prefer the existing implementation which is cleaner and consistent with other types.

Reply:
The trade-off is that this would require a separate reader just for boolean values. I'm open to that if there are strong objections, but it seems to me that allowing the actual type to be fixed up depending on the value is a reasonable change that saves quite a bit of code. If you object to this, I can revert it and add the boolean-specific reader.

Reply:
Yeah. You are right. `BOOLEAN_TRUE` and `BOOLEAN_FALSE` physical types need special handling. When we shred them, they should be grouped together, considered as one type. 

And also we may need to have a type list {`NULL`, `BOOLEAN`, `INT8`, `INT16`, etc}, which is almost same as physical type list with BOOLEAN for `BOOLEAN_TRUE` and `BOOLEAN_FALSE`. Otherwise, we can't represent a shredded column type for true/false. 

I'm fine to keep it simple for now. We can revisit if needed.